### PR TITLE
Fixes #9461 - Update subnets on domain change in hostgroup

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

**Comment by tbrisker** [comment]
[test]

**Comment by ares** [comment]
Why did you remove it from partials? I see they were not used anywhere else, but still...

**Comment by tbrisker** [comment]
No point in leaving it in common, maybe make the entire network tab a partial under hostgroups? it made sense to have it in common when it was used for hosts as well, but now it makes no sense.

File: app/views/common/_domain.html.erb [hunk 1]
```diff
@@ -1,11 +0,0 @@
-<%= fields_for item do |f| %>
-  <%= select_f f, :subnet_id, domain_subnets, :id, :to_label,
-               { :include_blank => domain_subnets.any? ? blank_or_inherit_f(f, :subnet) : _("No subnets") },
-               { :disabled => domain_subnets.empty?,
-                 :data     => { :url     => SETTINGS[:unattended] ? freeip_subnets_path : '',
-                                :subnets => subnets_ipam_data.to_json },
-                 :label    => _("Subnet"),
-                 :help_inline => :indicator,
-                 :onchange => 'subnet_selected(this);' }
```
Comment (ares) [inline app/views/common/_domain.html.erb:9]: where did this go? wasn't it renamed to `interface_subnet_selected`

**Comment by ares** [comment]
Ah I see, they are now quite different for interface and host gorup. Ok, if we ever add multiple interfaces to host group we may reextract common bits. Thanks for clarification.

File: app/views/common/_domain.html.erb (hunk 1 above)
Comment (tbrisker) [inline app/views/common/_domain.html.erb:9]: If I understand correctly, `interface_subnet_selected` is only relevant for interfaces, not hostgroups (autoselect IP etc.). Am I missing something here?

Comment (ares) [inline app/views/common/_domain.html.erb:9]: dismiss, there's no point of calling this as we don't set any IP in hostgroup

**Comment by ares** [comment]
Tested, work fine, pending jenkins

**Comment by tbrisker** [comment]
[test]

**Comment by dLobatog** [comment]
@tbrisker @ares Jenkins is :traffic_light: :green_heart: !

**Comment by ares** [comment]
Merged as 3345d139a40ed461114455acf2f2f26facaa5d98, thanks @tbrisker!

**Pull request closed without merging**